### PR TITLE
Feature/bugfix tc forecast basins

### PR DESCRIPTION
**Pull request by manniepmkam**
Adding an extra basin 'X - Undefined Basin' to hold TC forecasts which has an unknown classification of basins from ECMWF, otherwise the method `TCForecast.fetch_ecmwf()` would be disrupted if there exist such TC in the bufr files. An example from [20210121000000 forecast](https://www.ecmwf.int/en/forecasts/charts/tcyclone/tc_plumes?facets=undefined&time=2021012100,0,2021012100&unique_id=XXX_XXX_2021).
LOGGER.info is printed out when there is a tracks belongs to the undefined basin.

### CODE REVIEW

File: climada/hazard/tc_tracks_forecast.py [hunk 1]
```diff
@@ -56,7 +56,8 @@
     'A': 'A - Arabian Sea (North Indian Ocean)',
     'B': 'B - Bay of Bengal (North Indian Ocean)',
     'U': 'U - Australia',
-    'S': 'S - South-West Indian Ocean'
+    'S': 'S - South-West Indian Ocean',
+    'X': 'X - Undefined Basin'
 }
 """Gleaned from the ECMWF wiki at
 https://confluence.ecmwf.int/display/FCST/Tropical+Cyclone+tracks+in+BUFR+-+including+genesis
```
Comment (ChrisFairless) [inline climada/hazard/tc_tracks_forecast.py:63]: I think it's worth extending this comment to explain where 'X' comes from

```suggestion
https://confluence.ecmwf.int/display/FCST/Tropical+Cyclone+tracks+in+BUFR+-+including+genesis
with added basin 'X' to deal with it appearing in operational forecasts
(see e.g. years 2020 and 2021 in the sidebar at https://www.ecmwf.int/en/forecasts/charts/tcyclone/)
```

**Comment by ChrisFairless** [comment]
looks good!

**Review by ThomasRoosli — APPROVED** [review]
Looks good for me. Great that you included an LOGGER.info, to draw attention to such tracks.

**Comment by mmyrte** [comment]
Thanks, Mannie! Looking good. As soon as I get my head working again, I'll commit some docstring changes for the module.

**Pull request merged**